src/librustc_mir_build/hair/pattern/check_match.rs

```diff
@@ -490,6 +490,14 @@ fn check_exhaustive<'p, 'tcx>(
          possibly by adding wildcards or more match arms",
     );
     err.note(&format!("the matched value is of type `{}`", scrut_ty));
+    if (scrut_ty == cx.tcx.types.usize || scrut_ty == cx.tcx.types.isize)
+        && joined_patterns == "`_`"
+    {
+        err.note("for `usize` and `isize`, no assumptions about the maximum value are permitted");
```

Comment:
How about only having a single note, with something like:
```
`usize` does not have a fixed maximum value, so a wildcard `_` is necessary  to match exhaustively
```

Reply:
Maybe it would also be good to add:
```rust
if nightly_options::is_nightly_build() {
    err.help(&format!("add `#![feature(precise_pointer_size_matching)]` to the crate attributes to enable precise `usize` matching", feature));
}
```

Reply:
It'd be even better to only trigger these warnings if the entire range was covered, but that's probably unnecessary.

Reply:
Addressed some reviews via b93ecc1dacd3d534b0fa24e4b14cd3b07bae9889 but I have no idea for the last point.